### ISSUE TITLE
Could maps be updated when data in external files changes?

### ISSUE THREAD

**Issue by claremacrae** [javalent/obsidian-leaflet#268]
I have some maps that specify their data from external files. For example:

```
\```leaflet
id: blah
geojson: [[blah-geojson.json]]
```

```
\```leaflet
id: csf-supermarkets
markerTag: CSF/Yes
```

```
\```leaflet
id: csf-supermarkets-all
markerFolder: Clare/Volunteering/CSF Locations
```

**Expected behaviour:**

When I edit data in the external files, the map should refresh

**Actual behaviour:**

I have to edit some parameter in the leaflet block, like the size, to force the refresh - and then undo the edit.

It would be wonderful if maps could be redrawn if data used by them changes.

**Comment by valentine195** [comment]
I’ll take a look - markerTag and markerFolder should be updating any markers when a file associated with them changes. 

GPX and GeoJSON does *not* register file watchers, although I could explore that.

**Comment by claremacrae** [comment]
OK, I'll try and create a set of repro steps or a video for the markerTag and markerFolder cases.

**Comment by claremacrae** [comment]
Here is a demo video. I'll upload the files in a later message.

![Leaflet refresh bug - take 2](https://user-images.githubusercontent.com/4840096/147842340-4b22299b-85b7-43be-ad54-46ecc9819d87.gif)

**Comment by claremacrae** [comment]
Here are the files used:

- [CSF Today - Map.md](https://github.com/valentine195/obsidian-leaflet-plugin/files/7797279/CSF.Today.-.Map.md)
- [Tesco 1.md](https://github.com/valentine195/obsidian-leaflet-plugin/files/7797280/Tesco.1.md)
- [Tesco 2.md](https://github.com/valentine195/obsidian-leaflet-plugin/files/7797281/Tesco.2.md)

**Comment by claremacrae** [comment]
This video was with Leaflet 4.8.0

**Comment by claremacrae** [comment]
One last thing for now.

After making the demo video, I quit Obsidian, and edited the two tags back to `CSF/Yes`
I then restarted Obsidian, and the markers did not reappear on the screen.

This is why I previously asked if there was some caching going on somewhere - as I can't otherwise work out why the map would not be re-drawn when I have quit and restarted Obsidian - and after that, I still have to edit a setting in the leaflet block in order to get the map to update.

**Comment by claremacrae** [comment]
Thanks. Could you say how or when this was completed @sigrunixia?

**Comment by sigrunixia** [comment]
> Thanks. Could you say how or when this was completed @sigrunixia?

Hi @claremacrae 

I had to double check with Javalent on this one, because my notes had this one as completed. However, it was a similar scenario with embedded markdown files that was resolved. I apologize for leading you to believe this was fixed when it wasn't! D:

I'll go ahead and re-open, and get this tagged so we can look at it in a near release.
